packages/toolpad-components/src/Text.tsx

```diff
@@ -23,9 +22,22 @@ interface TextProps extends Omit<BaseProps, 'children'> {
 
 const MarkdownContainer = styled('div')({
   display: 'block',
-  [`&:empty::before`]: { content: '""', display: 'inline-block' },
+  maxWidth: 'fill-available',
+  '& span:empty::before': {
+    content: '""',
+    display: 'inline-block',
+  },
 });
 
+const CodeContainer = styled('pre')(({ theme }) => ({
+  backgroundColor: theme.palette.grey[200],
+  marginLeft: theme.spacing(1),
+  marginRight: theme.spacing(1),
+  borderRadius: theme.shape.borderRadius,
+  padding: theme.spacing(1),
+  overflowX: 'scroll',
```

Comment:
overflow scroll is very rarely helpful because it always shows the scrollbar, e.g.

<img width="558" alt="Screenshot 2023-01-09 at 00 47 07" src="https://user-images.githubusercontent.com/3165635/211225002-74420aff-7761-407e-8023-f92bc727605e.png">

So:

```suggestion
  overflow: 'auto',
```